packages/kbn-scout/src/config/schema/schema.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import Joi from 'joi';
+
+const maybeRequireKeys = (keys: string[], schemas: Record<string, Joi.Schema>) => {
+  if (!keys.length) {
+    return schemas;
+  }
+
+  const withRequires: Record<string, Joi.Schema> = {};
+  for (const [key, schema] of Object.entries(schemas)) {
+    withRequires[key] = keys.includes(key) ? schema.required() : schema;
+  }
+  return withRequires;
+};
+
+const urlPartsSchema = ({ requiredKeys }: { requiredKeys?: string[] } = {}) =>
+  Joi.object()
+    .keys(
+      maybeRequireKeys(requiredKeys ?? [], {
+        protocol: Joi.string().valid('http', 'https').default('http'),
+        hostname: Joi.string().hostname().default('localhost'),
+        port: Joi.number(),
+        auth: Joi.string().regex(/^[^:]+:.+$/, 'username and password separated by a colon'),
+        username: Joi.string(),
+        password: Joi.string(),
+        pathname: Joi.string().regex(/^\//, 'start with a /'),
+        hash: Joi.string().regex(/^\//, 'start with a /'),
+        certificateAuthorities: Joi.array().items(Joi.binary()).optional(),
+      })
+    )
+    .default();
+
+const requiredWhenEnabled = (schema: Joi.Schema) => {
+  return Joi.when('enabled', {
+    is: true,
+    then: schema.required(),
+    otherwise: schema.optional(),
+  });
+};
+
+const dockerServerSchema = () =>
+  Joi.object()
+    .keys({
+      enabled: Joi.boolean().required(),
+      image: requiredWhenEnabled(Joi.string()),
+      port: requiredWhenEnabled(Joi.number()),
+      portInContainer: requiredWhenEnabled(Joi.number()),
+      waitForLogLine: Joi.alternatives(Joi.object().instance(RegExp), Joi.string()).optional(),
+      waitForLogLineTimeoutMs: Joi.number().integer().optional(),
+      waitFor: Joi.func().optional(),
+      args: Joi.array().items(Joi.string()).optional(),
+    })
+    .default();
+
+export const schema = Joi.object()
+  .keys({
+    serverless: Joi.boolean().default(false),
+
+    // users: Joi.object().pattern(
```

Comment:
looks like we are not using this in the config for now, so it can be removed

Reply:
updated in 45c497444aaf7223974ddc615c598ee88b84ed8e